src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectDependenciesCommandResolver.cs

```diff
@@ -97,12 +97,18 @@ public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
         private LockFileTargetLibrary GetToolLibraryForContext(
             ProjectContext projectContext, string commandName)
         {
-            var toolLibrary = projectContext.LockFile.Targets
+            var toolLibraries = projectContext.LockFile.Targets
                 .FirstOrDefault(t => t.TargetFramework.GetShortFolderName()
                                       .Equals(projectContext.TargetFramework.GetShortFolderName()))
-                ?.Libraries.FirstOrDefault(l => l.Name == commandName);
+                ?.Libraries.Where(l => l.Name == commandName ||
```

Comment:
It's just cheaper not to go inside if you don't have to but I can remove the first case.